### PR TITLE
Add Percona as a database server option

### PR DESCRIPTION
**Pull request by dlundgren**
This allows Percona to be chosen as an alternative to MariaDB.

### CODE REVIEW

**Comment by QWp6t** [comment]
Have you tested provisioning with mariadb and then provisioning the same server with percona, and vice versa? Should one be completely uninstalled before installing the other?

**Comment by dlundgren** [comment]
I have not tested that particular scenario. MariaDB & Percona are mutually exclusive and replace/break each other.

I should be able to take a look at having it detect which is installed, and handling that scenario, later this week.

**Comment by swalkinshaw** [comment]
These changes seem pretty minor so I'd be open to adding this. Can the duplication in the provider tasks be minimized? Maybe just include a task file for provider install which would only have the first ~4 tasks to actually install it.

**Comment by dlundgren** [comment]
I moved the block into the main task, and left the installers in their own files.

I also noticed an issue with which release I was installing from when using `percona_dist`, I dropped that in favor of using `ansible_distribution_release`.

File: roles/mysql-server/defaults/main.yml [hunk 1]
```diff
@@ -1,8 +1,15 @@
-mariadb_binary_logging_disabled: true
+# change this to percona if you want to use percona
```
Comment (swalkinshaw) [inline roles/mysql-server/defaults/main.yml]: Maybe: `# providers: mariadb or percona`

File: dev.yml [hunk 2]
```diff
@@ -1,16 +1,15 @@
 ---
-- name: "WordPress Server: Install LEMP Stack with PHP 7.0 and MariaDB MySQL"
+- name: "WordPress Server: Install LEMP Stack with PHP 7.0 and chosen MySQL"
```
Comment (swalkinshaw) [inline dev.yml]: While you're here, let's just simplify this to: "WordPress Server: Install LEMP stack"

File: server.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@
     - name: Install Python 2.x
       raw: sudo apt-get install -qq -y python-simplejson
 
-- name: WordPress Server - Install LEMP Stack with PHP 7.0 and MariaDB MySQL
+- name: "WordPress Server - Install LEMP Stack with PHP 7.0 and chosen MySQL"
```
Comment (swalkinshaw) [inline server.yml]: Same here

File: roles/mysql-server/tasks/percona.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Add Percona MySQL apt-key
+  apt_key:
+    url: "http://keyserver.ubuntu.com/pks/lookup?op=get&fingerprint=on&search={{ percona_keyserver_fingerprint }}"
```
Comment (swalkinshaw) [inline roles/mysql-server/tasks/percona.yml]: Let's use `https` here

File: roles/mysql-server/tasks/mariadb.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Add MariaDB MySQL apt-key
+  apt_key:
+    url: "http://keyserver.ubuntu.com/pks/lookup?op=get&fingerprint=on&search={{ mariadb_keyserver_fingerprint }}"
```
Comment (swalkinshaw) [inline roles/mysql-server/tasks/mariadb.yml]: And `https` here as well :)

File: roles/mysql-server/defaults/main.yml [hunk 6]
```diff
@@ -1,8 +1,15 @@
...
+
 mariadb_keyserver_fingerprint: "0xcbcb082a1bb943db"
 mariadb_mirror: nyc2.mirrors.digitalocean.com
 mariadb_version: "10.0"
 mariadb_dist: trusty
+
+percona_keyserver_fingerprint: "0x9334A25F8507EFA5"
+percona_version: "5.7"
+
+binary_logging_disabled: true
```
Comment (swalkinshaw) [inline roles/mysql-server/defaults/main.yml]: `mysql_binary_logging_disabled`

**Comment by swalkinshaw** [comment]
I'll test this out but otherwise it should be good to go 👍

**Comment by swalkinshaw** [comment]
@dlundgren I'm sorry about this but I've changed my mind about including this in Trellis :( Mostly I don't want to incur the overhead of supporting both MariaDB and Percona. It might not be much, but it's obviously more than just MariaDB alone and increases the support/maintenance burden. This is also the first request of Percona we've had in ~2 years and anyone who needs it can easily add it themselves.

I also hadn't thought about what would happen once we switched to the official Ubuntu distro packages for Maria in https://github.com/roots/trellis/pull/693. Now that the custom apt repo is gone, these two database options diverge just a little more.

Thanks for the good work on this and sorry for not deciding this sooner before I got you to do additional work.

**Comment by dlundgren** [comment]
That's fine, would you consider the option to allow mysql vs maria using distro packages?

**Pull request closed without merging**